cpp/src/arrow/compute/exec/aggregate.cc

```diff
@@ -61,7 +61,10 @@ Result<std::vector<std::unique_ptr<KernelState>>> InitKernels(
       // use known default options for the named function if possible
       auto maybe_function = ctx->func_registry()->GetFunction(aggregates[i].function);
       if (maybe_function.ok()) {
-        options = maybe_function.ValueOrDie()->default_options();
+        FunctionOptions* default_opts = maybe_function.ValueOrDie()->default_options();
```

Comment:
Hmm, I guess this doesn't work. Maybe `std::move(maybe_function).MoveValueUnsafe()->default_options()` or something?